.github/workflows/release.yaml

```
@@ -16,8 +16,63 @@ permissions:
   packages: write
 
 jobs:
+  setup:
+    name: Setup Release
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.set_vars.outputs.version }}
+      goreleaser_args: ${{ steps.set_vars.outputs.goreleaser_args }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set the release related variables
+        id: set_vars
+        run: |
+          # run the full release pipeline unless it's a pull request
+          if [[ ${{ github.event_name }} == 'workflow_dispatch' ]]; then
+            VERSION="$(git describe --abbrev=0 --tags)"
+            echo "goreleaser_args=--clean --skip=validate" >> $GITHUB_OUTPUT
+          elif [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION="${GITHUB_REF#refs/tags/}"
+            echo "goreleaser_args=--clean" >> $GITHUB_OUTPUT
+          elif [[ $GITHUB_REF == refs/pull/* ]]; then
+            VERSION="$(git describe --abbrev=0 --tags)+build.$(echo "${GITHUB_REF}" | sed -E 's|refs/pull/([^/]+)/?.*|\1|')"
+            echo "goreleaser_args=--snapshot --clean" >> $GITHUB_OUTPUT
+          else
+            VERSION="$(git describe --abbrev=0 --tags)"
+            echo "goreleaser_args=--snapshot --clean" >> $GITHUB_OUTPUT
+          fi
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+
+  helm:
+    name: Package Helm Charts
+    needs: setup
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Login to GitHub Container Registry
+        if: ${{ github.event_name != 'pull_request' }}
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | helm registry login ${{ env.IMAGE_REGISTRY }} -u ${{ github.repository_owner }} --password-stdin
+
+      - name: Lint kgateway chart
+        run: helm lint install/helm/kgateway
```

run this first? so if it fails, it fails fast?

---

actually, this should run on PRs; not on release